pkg/driver/camera/camera_linux.go

```diff
@@ -314,13 +316,18 @@ func (c *camera) Properties() []prop.Media {
 			}
 
 			if frameSize.StepWidth == 0 || frameSize.StepHeight == 0 {
-				properties = append(properties, prop.Media{
-					Video: prop.Video{
-						Width:       int(frameSize.MaxWidth),
-						Height:      int(frameSize.MaxHeight),
-						FrameFormat: supportedFormat,
-					},
-				})
+				for _, framerate := range c.cam.GetSupportedFramerates(format, frameSize.MinWidth, frameSize.MinHeight) {
+					for _, fps := range enumFramerate(framerate) {
+						properties = append(properties, prop.Media{
+							Video: prop.Video{
+								Width:       int(frameSize.MaxWidth),
+								Height:      int(frameSize.MaxHeight),
+								FrameFormat: supportedFormat,
+								FrameRate:   fps,
+							},
+						})
+					}
+				}
 			} else {
 				// FIXME: we should probably use a custom data structure to capture all of the supported resolutions
 				for _, supportedResolution := range supportedResolutions {
```

Comment:
I know you didn't write this code but just curious if you know why this lib would limit the resolutions to the ones in `supportedResolutions` instead of allowing everything the camera returns. It isn't obvious to me at all.

Reply:
Yes I think we can safely handle stepwise `FrameSize` returns without hardcoded list. 

Here is the [PR](https://github.com/pion/mediadevices/commit/16bcd0b7dd6932e2f626c6bcf05192480e908978) that added list.

This is also relevant to stepwise FrameRate returns - may need to revisit the data-structures for cleaner stepwise handling.

Reply:
Ah, yeah I totally misread the FIXME. Looks like someone planned on doing exactly that. Probably isn't a big deal either for now.